src/WebservConfig/src/WebservConfig.hpp

```diff
@@ -0,0 +1,42 @@
+#ifndef WEBSERVCONFIG_HPP
+#define WEBSERVCONFIG_HPP
+
+#include <list>
+
+#include "../../Sys/src/File.hpp"
+#include "../../Sys/src/BufReader.hpp"
+#include "../../Str/src/Str.hpp"
+
+namespace Webserv {
+
+class ConfigStore
+{
+	public:
+		typedef enum { FAILED, OK, INVALID_FILE } ParseStatus;
+		typedef Utils::result<ParseStatus, std::string> ParseResult;
```

Comment:
I like the enum here, but I think the Result could be used even more effectively here - combining construction and parsing into a single step using a named constructor:

```suggestion
                typedef enum { FAILED, OK, INVALID_FILE } ParseStatus;
		typedef Utils::result<ConfigStore, ParseStatus> Result;
		
		static Result create_from_file(const char* fname);
		static ConfigStore create_default(void); // default named constructor - default configuration, cannot fail
```

Then in the codebase and testing:
```
     ConfigStore::Result config_res = ConfigStore::create_from_file(config.file);
     
     // Testing if parsing of 'config.file' was successful
     EXPECT_TRUE(config_res.is_ok());

     // After verifying that parsing was successful, we unwrap the value
    ConfigStore config = config_res.unwrap();
    
    // Now, combined with a private default constructor for ConfigStore, it should be literally impossible to create an invalid instance of ConfigStore - you can only create it via 'create_from_file', where all error handling for parse errors is placed, or with 'create_default', which creates a default ConfigStore
```